hunit/Test/Tasty/HUnit/Steps.hs

```diff
@@ -18,25 +21,46 @@ instance IsTest TestCaseSteps where
 
     let
       stepFn :: String -> IO ()
-      stepFn msg = atomicModifyIORef ref (\l -> (msg:l, ()))
+      stepFn msg = do
+        tme <- getTime
+        atomicModifyIORef ref (\l -> ((tme,msg):l, ()))
 
-    hunitResult <- try (assertionFn stepFn)
+    hunitResult <- (Right <$> assertionFn stepFn) `catches`
+      [ Handler (\(HUnitFailure mbloc errMsg) -> return $ Left (prependLocation mbloc errMsg))
+      , Handler (\(SomeException ex)          -> return $ Left (show ex))
+      ]
 
-    msgs <- reverse <$> readIORef ref
+    endTime <- getTime
+
+    msgs <- snd . foldl'
```

Comment:
You use a strict fold on a tuple parameter. It won't do much — it will just force the tuple constructor. To avoid creating thunks for duration/duration', you need to force them (e.g. by using bang patterns).

Reply:
I didn’t think of it much, I just needed a left fold, so I chose this one kinda automatically. :sweat_smile: 

So instead of
```haskell
(curTime, (msg ++ duration'):acc)
```
should I use
```haskell
(curTime,) $! (msg ++ duration'):acc
```
or even
```haskell
(curTime,) $! ((: acc) $! (msg ++ duration'))
```
?

Reply:
No, more like ```duration' `seq` (curTime, (msg ++ duration'):acc))```, although I'd prefer bang patterns:

```
let !duration = ...
    !duration' = ...
in ...
```

Reply:
Okay.